include/rdma/fi_eq.h

```diff
@@ -239,6 +239,8 @@ struct fi_cq_err_entry {
 	/* err_data is available until the next time the CQ is read */
 	void			*err_data;
 	size_t			err_data_size;
+	void			*auth_key;
+	size_t			auth_key_size;
```

Comment:
If FI_AUTH_KEY_MATCH_ALL is removed (thus ensuring a fi_addr_t exists for each auth key), I am not sure if we need to have `fi_cq_err_entry::auth_key` and `fi_cq_err_entry::auth_key_size` as currently defined. Seems like we can just defined `fi_addr_t fi_cq_err_entry::auth_key` and return the `fi_addr_t` from insert. Thoughts?

Reply:
I think you're right, and you're also exposing a gap in the current API.  We can't report the src_addr with any error completion, which may be useful.  For example, if we report a truncated receive completion at the target, we can't report the source of the message.

So... I like the change you're proposing, but I'd rename the 'auth_key' field to 'src_addr' to make it more generic.  The app would need to use the error code to know to interpret the src_addr as an auth_key only address.

Reply:
Sounds good. Do you want me to open a new issue for truncated recvs not reporting fi_addr_t?

Reply:
You can if you want; it seems like a gap that should be addressed, which is easier to handle with your proposal.